### ISSUE TITLE
boost boost

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#2799]
With x86 builds gone we could switch to a newer version of boost, which seems to be 1.71.0 for 20.04.

**Comment by mloskot** [comment]
BTW, what is the status of the Boost requirement now?


The `filesystem` and `system` are gone and just the `program_options` seems required for the data tools build:

https://github.com/valhalla/valhalla/blob/f005efa91e85e8f1df10e28c041ea8d35bed477c/CMakeLists.txt#L220-L222

However, come the CMake on Windows CI where we do request the data tools build

https://github.com/valhalla/valhalla/blob/f005efa91e85e8f1df10e28c041ea8d35bed477c/.azure-pipelines.yml#L112

does not report, e.g. here https://dev.azure.com/valhalla1/valhalla/_build/results?buildId=9377&view=logs&j=8c5a3053-cc94-5e24-da52-c43aabf65e0e&t=de4ba696-e3b4-5027-1054-4f8c14f9b343, anything like

```
-- Found Boost: <BOOST_ROOT>/boost/boost (found suitable version "1.76.0", minimum required is "1.51") found components: program_options
```

AFAIU, Valhalla still requires lots of headers (referenced via CMake target `Boost::boost`).

**Comment by nilsnolde** [comment]
> does not report

Could that be the stripped message bcs it's caching the build directory? It's not reporting a lot of stuff actually.. It builds tiles on Win CI, so that works. But yeah, program_options seems to be the only lib component it uses these days, rest header-only AFAIK (boost:geometry, boost:optional etc).

I think I opened this issue rather for the minimum required version, back then I thought I'd need some bg stuff that wasn't available in older versions. 

Are you asking bcs you'd like to bump the min version or just general?

**Comment by mloskot** [comment]
Right, I missed to take the caching into play. Still not sure how that could make CMake report stuff selectively - CMake runs anyway.

I'm asking because I noticed some strange failures when building the data tools from Valhalla 3.0, 
for example, `src/mjolnir`  is failing due to missing Boost headers - it does not explicitly 'link' to `Boost::boost`:

https://github.com/valhalla/valhalla/blob/f005efa91e85e8f1df10e28c041ea8d35bed477c/src/mjolnir/CMakeLists.txt#L84-L92

It seems Boost headers are dragged via https://github.com/valhalla/valhalla/blob/f005efa91e85e8f1df10e28c041ea8d35bed477c/src/baldr/CMakeLists.txt#L117-L123

but for some not yet known reason it does not seem to work for me.
So, I looked at our Windows CI

----

I'm going to upgrade to the latest Valhalla and see.

**Comment by nilsnolde** [comment]
> Still not sure how that could make CMake report stuff selectively - CMake runs anyway.

If the CMake cache is there I always observe the same, not even half the reporting of a clean cmake run :shrug: 

I think the earliest I tried on Win was 3.0.9, but I remember a lot of stuff was shuffled around in the CMakeLists due to selective failures in the past year at least

**Comment by kevinkreiser** [comment]
Yep, sadly as @mloskot mentioned we still use a lot of header only boost deps in the library and we link program options in many executables. We definitely could shed the latter but it would take some work for sure

**Comment by mloskot** [comment]
I've continued the CMake issues above as the discussion thread here https://github.com/valhalla/valhalla/discussions/3105

**Comment by nilsnolde** [comment]
this can be closed now, superseded by #3346
